julia/ducc0.jl

```diff
@@ -302,23 +300,91 @@ function sht_alm2leg(
     nthreads,
     Desc(leg),
   )
+  end
   if ret != 0
     throw(error())
+  else
+    leg
   end
 end
 
-function sht_leg2map(
+function sht_alm2leg(
+  alm::StridedArray{Complex{T},2},
+  spin::Unsigned,
+  lmax::Unsigned,
+  mval::StridedArray{Csize_t,1},
+  mstart::StridedArray{Csize_t,1}, # FIXME: this is still 0-based at the moment
+  lstride::Int,
+  theta::StridedArray{Cdouble,1},
+  nthreads::Unsigned,
+) where {T}
+  ncomp = size(alm, 2)
+  ntheta = length(theta)
+  nm = length(mval)
+  leg = Array{Complex{T}}(undef, (nm, ntheta, ncomp))
+  sht_alm2leg!(alm, leg, spin, lmax, mval, mstart, lstride, theta, nthreads)
+end
+
+function sht_leg2alm!(
+  leg::StridedArray{Complex{T},3},
+  alm::StridedArray{Complex{T},2},
+  spin::Unsigned,
+  lmax::Unsigned,
+  mval::StridedArray{Csize_t,1},
+  mstart::StridedArray{Csize_t,1}, # FIXME: this is still 0-based at the moment
+  lstride::Int,
+  theta::StridedArray{Cdouble,1},
+  nthreads::Unsigned,
+) where {T}
+  GC.@preserve leg mval mstart theta alm begin
+  ret = ccall(
+    (:sht_leg2alm, libducc),
+    Cint,
+    (Dref, Csize_t, Csize_t, Dref, Dref, Cptrdiff_t, Dref, Csize_t, Dref),
+    Desc(leg),
+    spin,
+    lmax,
+    Desc(mval),
+    Desc(mstart),
+    lstride,
+    Desc(theta),
+    nthreads,
+    Desc(alm),
+  )
+  end
+  if ret != 0
+    throw(error())
+  else
+    alm
+  end
+end
+
+function sht_leg2alm(
   leg::StridedArray{Complex{T},3},
+  spin::Unsigned,
+  lmax::Unsigned,
+  mval::StridedArray{Csize_t,1},
+  mstart::StridedArray{Csize_t,1}, # FIXME: this is still 0-based at the moment
+  lstride::Int,
+  theta::StridedArray{Cdouble,1},
+  nthreads::Unsigned,
+) where {T}
+  ncomp = size(leg, 3)
+  nm = length(mval)
+  alm = Array{Complex{T}}(undef, (nm, ncomp))
```

Comment:
I think this should be
```
alm = Array{Complex{T}}(undef, (mstart[size(mstart,1)]+lmax+1, ncomp))
```
but I might be off by 1. Just `nm` as size is too small.